### ISSUE TITLE
Can't access {item} in onPress function

### ISSUE THREAD

**Issue by callyb** [jemise111/react-native-swipe-list-view#417]
Hi,

Thanks for all your work - great component!

I was using the experimental 'SwipeableFlatList' from react-native before until it was deprecated and with that package I could easily access data.item in each list item to use in the onPress function - I just passed {item} as an argument to the function attached to onPress (the same way it's passed to 'renderItem') then I could pass it to react-navigation as props to pass to the next screen so that only the list item's data appears in that screen (so it can be edited).  When I try that in react-native-swipe-list-view, I get "item is undefined". 

I feel like I'm  missing something?!  I've seen something about refs but don't understand how they would change access to {item}...any pointers would be much appreciated...thank you!

**Comment by StevenBlack** [comment]
Taking a guess here...

Since you're using `react-navigation`, for the `item` coming-in from the parent screen:

```js
    const { navigation } = this.props;
    const item = navigation.getParam('item');
```

**Comment by callyb** [comment]
Ok, I'm such a ditz!   I had the child component working fine when I was using 'SwipeableFlatList ' with `this.props.navigation.state.params.item` (same result as your suggestion) but thank you StevenBlack for getting me thinking about this another way round!  What I realised was that the problem was with the way I was not dealing with the fact that react-native-swipe-list-view uses `rowData.item` not just plain `item`...so all I needed to do was to change the onPress function (in the Parent component) from

`this.props.navigation.navigate("ChildScreen", { item })}` 
to
`this.props.navigation.navigate("ChildScreen", { item: rowData.item })}`

and now the whole thing is working fine.  (And yes, I have to update the version of react-navigation I'm using (v2) but it's not the top of my list right now!!)

Sorry for the detour - but maybe my brain fog moment might help someone else one day!!
